baseapp/abci.go

```diff
@@ -903,11 +914,20 @@ func (app *BaseApp) checkHalt(height int64, time time.Time) error {
 		halt = true
 	}
 
-	if halt {
-		return fmt.Errorf("halt per configuration height %d time %d", app.haltHeight, app.haltTime)
+	if !halt {
+		return nil
 	}
-
-	return nil
+	// [AGORIC] Make a best-effort attempt to kill our process.
+	p, err := os.FindProcess(os.Getpid())
+	if err != nil {
```

Comment:
The `p.Signal` calls that follow must only be done if `FindProcess` succeeds.

```suggestion
	if err == nil {
		// If we successfully found our own process, then
```

Reply:
yeah good catch